modin/experimental/sql/__init__.py

```diff
@@ -13,14 +13,62 @@
 
 import warnings
 
-try:
-    from dfsql import sql_query as query
-    import dfsql.extensions  # noqa: F401
-except ImportError:
-    warnings.warn(
-        "Modin experimental sql interface requires dfsql to be installed."
-        + " Run `pip install modin[sql]` to install it."
-    )
-    raise
-
-__all__ = ["query"]
+import modin.pandas as pd
+import modin.config as cfg
+
+
+_query_impl = None
+
+
+def query(sql: str, *args, **kwargs) -> pd.DataFrame:
+    """
+    Execute SQL query using either hdk_query or dfsql.
+
+    Parameters
+    ----------
+    sql : str
+        SQL query to be executed.
+    *args : *tuple
+        Positional arguments, passed to the execution engine.
+    **kwargs : **dict
+        Keyword arguments, passed to the execution engine.
+
+    Returns
+    -------
+    modin.pandas.DataFrame
+        Execution result.
+    """
+    global _query_impl
+
+    if _query_impl is None:
+        if cfg.StorageFormat.get() == "Hdk":
+            from modin.experimental.sql.hdk.query import hdk_query
+
+            _query_impl = hdk_query
+        else:
+            try:
+                from dfsql import sql_query
+
+                _query_impl = sql_query
```

Comment:
```suggestion
                from dfsql import sql_query as _query_impl
```
also voting for what @YarShev suggests